src/ReactiveDomain.Identity/Domain/Client.cs

```diff
@@ -0,0 +1,96 @@
+﻿using ReactiveDomain.Identity.Messages;
+using ReactiveDomain.Util;
+using System;
+using static ReactiveDomain.Identity.Messages.ClientMsgs;
+
+namespace ReactiveDomain.Identity.Domain
+{
+
+    public class Client : AggregateRoot
+    {
+        public string ClientName => _clientName;
+        private string _clientName;
+        private Client()
+        {
+            RegisterEvents();
+        }
+
+        private void RegisterEvents()
+        {
+            Register<ClientCreated>(@event => { Id = @event.ClientId; _clientName = @event.ClientName; });
+            Register<ClientSecretAdded>(@event => { });
+            Register<ClientSecretRemoved>(@event => { });
+        }
+
+        //constructor for Elbe client
+        public Client(
+                Guid id,
+                Guid applicationId,
+                string clientName, //todo: value object?
+                string encryptedClientSecret)
+        {
+            Ensure.NotEmptyGuid(id, nameof(id));
+            Ensure.NotEmptyGuid(applicationId, nameof(applicationId));
+            Ensure.NotNullOrEmpty(clientName, nameof(clientName));
+            Ensure.NotNullOrEmpty(encryptedClientSecret, nameof(encryptedClientSecret));
+            //todo: move url definitions into Elbe when adding the secret store
```

Comment:
So this whole ctor will go away at that point, correct?

Reply:
They will be passed in from the app at that point.

Reply:
So it'll use the other ctor and this one can go away. 🙂